packages/lisk-api-client/src/api_method.ts

```diff
@@ -12,20 +12,33 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-
+import { AxiosRequestConfig } from 'axios';
 import { GET } from './constants';
+import {
+	ApiHandler,
+	ApiResponse,
+	HashMap,
+	RequestConfig,
+	Resource,
+} from './types/types';
 import { solveURLParams, toQueryString } from './utils';
 
 // Bind to resource class
-const apiMethod = ({
-	method = GET,
-	path = '',
-	urlParams = [],
-	validator = null,
-	defaultData = {},
-	retry = false,
-} = {}) =>
-	function apiHandler(...args) {
+export const apiMethod = (
+	{
+		method = GET,
+		path = '',
+		urlParams = [],
+		validator,
+		defaultData = {},
+		retry = false,
+	}: RequestConfig = { method: GET },
```

Comment:
I don't understand what is happening here. Are we setting the default values twice for `method` property?

Reply:
Without it `#apiMethod: when no parameters are passed` test suite fails, because if there is no object passed at all it does not set default method to GET. Maybe this test is not necessary since apiMethod is not exposed and always called with at least { method: GET }? @shuse2